enhanced simularium url and firebase recipe path validation

rugeli opened the pull request:
Problem
=======
What is the problem this work solves, including
1. query parameters such as `AWSAccessKeyId` were not being stripped from the url when opening in a new tab 
2. closes #212

Solution
========
What I/we did to solve this problem
- removed query params from the url not only in firebase but also when opening a new tab
- a new field has been added to `recipes/` in firebase, making it easier to copy the firebase recipe path for packing
<img width="557" alt="Screenshot 2024-02-12 at 4 31 27 PM" src="https://github.com/mesoscope/cellpack/assets/91452427/52bda2d8-4db9-4650-a6a4-bae21ef74232">


- enhanced validation for the input path to check each part of the path and provided with corresponding error messages  



## Type of change
Please delete options that are not relevant.
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
We have tested several cases:
- fetching a recipe from an incorrect database (e.g. uploading recipe A to `dev` but fetching it from `staging`)
- misnaming any part of the path including database name, collection name or recipe id  

some of the expected behaviors: 
- run `pack -r wrong_db_name:recipes/one_sphere_v_1.0.0`, get error:
<img width="849" alt="Screenshot 2024-02-12 at 4 20 12 PM" src="https://github.com/mesoscope/cellpack/assets/91452427/9cfc9068-0cab-4254-aa46-2d93f67ef6c3">

- run `pack -r firebase:wrong_collection_name/one_sphere_v_1.0.0`, get error:
<img width="1176" alt="Screenshot 2024-02-12 at 4 21 42 PM" src="https://github.com/mesoscope/cellpack/assets/91452427/bf1a4468-5a55-4380-b451-645300e703b3">

- run `pack -r firebase:recipes/wrong_id` or choosing the wrong database, get error:
<img width="1634" alt="Screenshot 2024-02-12 at 4 22 31 PM" src="https://github.com/mesoscope/cellpack/assets/91452427/2336948e-c43d-412e-abca-378194000d91">

mogres commented on cellpack/autopack/AWSHandler.py:
```diff
@@ -89,10 +89,11 @@ def create_presigned_url(self, object_name, expiration=3600):
         file_name = self.upload_file(file_path)
         if file_name:
-            return file_name, self.create_presigned_url(file_name)
+            base_url = self.create_presigned_url(file_name).split("?")[0]
```
Nice!

meganrm replied:
I might push this into the the `create_presigned_url` function so the function returns a url that the next function knows it can use without any adjustments. I also think this is a good bug for some tests. Ie make a list of urls that you know would be a problem and make sure we catch all of them.

mogres reviewed (approved):
Looks good!

rugeli commented:
Recent changes since the last review:
- added function `is_url_valid` within `AWSHandler` class
- added a new test file: `cellpack/tests/test_aws_handler.py`

mogres reviewed (approved):
Thanks for adding tests!
